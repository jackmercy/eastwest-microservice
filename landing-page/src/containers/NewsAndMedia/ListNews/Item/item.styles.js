import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    display: 'flex',
    position: 'relative',
    '& + $root': {
      marginTop: rem(90),
    },
    '&:nth-child(2n + 2)::before': {
      backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.1),
    },
    '&::before': {
      content: '" "',
      left: '50%',
      zIndex: -1,
      top: rem(-16),
      position: 'absolute',
      width: '8.75em',
      height: '8.75em',
      borderRadius: rem(5),
      backgroundColor: hexToRgbA(COLORS.primary, 0.1),
      transform: 'translateX(-35%) rotate(45deg)',
    },
  },
  thumbnailWrapper: {
    borderRadius: rem(5),
    overflow: 'hidden',
    position: 'relative',
    width: '100%',
    flex: '50% 0 0',
    maxWidth: rem(400),
    height: rem(267),
  },
  thumbnail: {
    width: '100%',
    top: '50%',
    left: '50%',
    position: 'absolute',
    transform: 'translate(-50%, -50%)',
  },
  content: {
    marginLeft: rem(24),
    flex: '50%',
    display: 'flex',
    flexDirection: 'column',
  },
  category: {
    fontWeight: 600,
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.primary,
    margin: 0,
    textDecoration: 'none',
    '&:hover': { opacity: 0.8 },
  },
  title: {
    flex: '1 1 auto',
    fontWeight: 900,
    fontSize: rem(18),
    lineHeight: rem(28),
    marginTop: 0,
    marginBottom: rem(8),
    textDecoration: 'none',
    color: COLORS.textBlack01,
    '&:hover': { color: COLORS.primary },
  },
  description: {
    fontWeight: 400,
    fontSize: rem(16),
    lineHeight: rem(24),
    marginTop: 0,
    marginBottom: rem(16),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    root: {
      display: 'block',
      '& + $root': { marginTop: rem(32) },
      '&::before': {
        left: 0,
        top: '50%',
        width: '5.25em',
        height: '5.25em',
        transform: 'translateX(-50%) rotate(45deg)',
      },
    },
    thumbnailWrapper: { maxWidth: '100%', height: rem(232) },
    thumbnail: { width: '125%', height: 'auto' },
    content: { marginLeft: 0, marginTop: rem(16), display: 'block' },
  },
});
