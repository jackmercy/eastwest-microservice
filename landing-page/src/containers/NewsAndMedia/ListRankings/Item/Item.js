import Link from 'next/link';
import { memo, useMemo, useRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import LinesEllipsis from 'react-lines-ellipsis';
import responsiveHOC from 'react-lines-ellipsis/lib/responsiveHOC';

import { API_ENDPOINT } from '../../../../../configs';

import useStyles from './item.styles';

const ResponsiveEllipsis = responsiveHOC()(LinesEllipsis);

const Item = ({ data, index, dataTest, className }) => {
  const imgRef = useRef();
  const classes = useStyles();
  const href = useMemo(
    () => ({
      category: `/news-and-media/${data.category.slug}`,
      details: `/news-and-media/${data.category.slug}/${data.slug}`,
    }),
    [data],
  );

  return (
    <div className={classNames(classes.root, className)} data-test={dataTest}>
      <div className={classes.thumbnailLayout}>
        <div className={classes.thumbnailWrapper}>
          <Link href={href.details} passHref>
            <a href={href.details}>
              <img
                ref={imgRef}
                className={classes.thumbnail}
                alt={data.title}
                src={`${API_ENDPOINT}/files/public/${data.thumbnail}`}
                onError={() => {
                  if (imgRef.current) {
                    imgRef.current.src =
                      '/static/images/logo/east-west-no-text.svg';
                  }
                }}
                data-test="item-thumbnail"
                loading="lazy"
              />
            </a>
          </Link>
        </div>
        <p className={classes.ranking} data-test="item-index">
          0<span>{index}</span>
        </p>
      </div>
      <div className={classes.content}>
        <Link href={href.category} passHref>
          <a
            href={href.category}
            className={classes.category}
            data-test="item-category"
          >
            {data.category.name}
          </a>
        </Link>
        <Link href={href.details} passHref>
          <a
            href={href.details}
            className={classes.title}
            title={data.title}
            data-test="item-title"
          >
            <ResponsiveEllipsis
              text={data.title}
              maxLine="3"
              ellipsis="..."
              trimRight
              basedOn="words"
            />
          </a>
        </Link>
      </div>
    </div>
  );
};
Item.propTypes = {
  data: PropTypes.shape({
    slug: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    category: PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
    }).isRequired,
    thumbnail: PropTypes.string.isRequired,
  }).isRequired,
  index: PropTypes.number.isRequired,
  dataTest: PropTypes.string,
  className: PropTypes.string,
};
Item.defaultProps = { dataTest: '', className: '' };

export default memo(Item);
