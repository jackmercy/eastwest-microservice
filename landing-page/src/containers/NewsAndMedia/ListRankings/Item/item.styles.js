import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: { display: 'flex', '& + $root': { marginTop: rem(48) } },
  thumbnailLayout: {
    flex: 'none',
    position: 'relative',
    width: rem(120),
    height: rem(120),
  },
  thumbnailWrapper: {
    borderRadius: rem(5),
    overflow: 'hidden',
    position: 'relative',
    width: '100%',
    height: '100%',
  },
  thumbnail: {
    width: '100%',
    top: '50%',
    left: '50%',
    position: 'absolute',
    transform: 'translate(-50%, -50%)',
  },
  ranking: {
    top: 0,
    left: '50%',
    position: 'absolute',
    transform: 'translate(-50%, -50%)',
    whiteSpace: 'nowrap',
    zIndex: 1,
    fontWeight: 900,
    fontSize: rem(64),
    lineHeight: rem(64),
    margin: 0,
    color: COLORS.textBlack,
    '& > span': { color: COLORS.primary },
  },
  content: {
    marginLeft: rem(24),
    flex: '1 1 auto',
    display: 'flex',
    flexDirection: 'column',
  },
  category: {
    fontWeight: 600,
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.primary,
    margin: 0,
    textDecoration: 'none',
    '&:hover': { opacity: 0.8 },
  },
  title: {
    flex: '1 1 auto',
    fontWeight: 900,
    fontSize: rem(18),
    lineHeight: rem(28),
    marginTop: rem(8),
    marginBottom: 0,
    textDecoration: 'none',
    color: COLORS.textBlack01,
    '&:hover': { color: COLORS.primary },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { '& + $root': { marginTop: rem(68) } },
    content: { marginLeft: rem(16) },
    title: { fontSize: rem(16), lineHeight: rem(24), marginTop: rem(14) },
  },
});
