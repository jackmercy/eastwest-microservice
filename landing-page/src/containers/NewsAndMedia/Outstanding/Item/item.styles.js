import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    height: '100%',
    overflow: 'hidden',
    position: 'relative',
    borderRadius: rem(5),
    '&.main $content::before': { backgroundColor: COLORS.primary },
  },
  thumbnail: {
    width: '100%',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    position: 'absolute',
    backdropFilter: 'blur(5px)',
  },
  content: {
    zIndex: 1,
    left: 0,
    bottom: 0,
    width: '100%',
    boxSizing: 'border-box',
    borderRadius: [rem(5), rem(5), 0, 0],
    position: 'absolute',
    padding: [rem(32), rem(40), rem(53)],
    backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.4),
    '&::before': {
      content: '" "',
      left: 0,
      zIndex: -1,
      top: rem(-32),
      position: 'absolute',
      width: '9.5em',
      height: '9.5em',
      borderRadius: rem(5),
      backgroundColor: COLORS.bgBlack01,
      transform: 'translateX(20%) rotate(45deg)',
    },
  },
  category: {
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    margin: 0,
    color: COLORS.textWhite,
    textDecoration: 'none',
    '&:hover': { opacity: 0.8 },
  },
  title: {
    fontWeight: 900,
    fontSize: rem(40),
    lineHeight: rem(56),
    overflow: 'hidden',
    marginTop: rem(16),
    marginBottom: 0,
    color: COLORS.textWhite,
    textDecoration: 'none',
    display: 'block',
    '&:hover': { opacity: 0.8 },
  },
  smallView: {
    '&.small': {
      '& $content': {
        padding: [rem(24), rem(24), rem(32)],
        '&::before': { top: rem(-8), width: '4.5em', height: '4.5em' },
      },
      '& $category': { fontSize: rem(12), lineHeight: rem(16) },
      '& $title': {
        fontSize: rem(18),
        lineHeight: rem(28),
        marginTop: rem(8),
      },
    },
  },

  [`@media only screen and (min-width: ${BREAKPOINTS.minMd}px)`]: {
    root: {
      '&.small': {
        '& $content': {
          padding: [rem(24), rem(24), rem(32)],
          '&::before': {
            top: rem(-8),
            width: '4.5em',
            height: '4.5em',
            backgroundColor: COLORS.bgBlack01,
          },
        },
        '& $category': { fontSize: rem(12), lineHeight: rem(16) },
        '& $title': {
          fontSize: rem(18),
          lineHeight: rem(28),
          marginTop: rem(8),
        },
      },
    },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    content: {
      padding: [rem(10), rem(16), rem(12)],
      minHeight: rem(103),
      '&::before': { top: rem(-8), width: '3em', height: '3em' },
    },
    category: { fontSize: rem(12), lineHeight: rem(16) },
    title: { fontSize: rem(18), lineHeight: rem(32), marginTop: 0 },
  },
});
