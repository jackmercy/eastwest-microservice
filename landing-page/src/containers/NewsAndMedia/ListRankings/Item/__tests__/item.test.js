import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Item from '../Item';

describe('Testing NewsAndMedia > ListRankings > Item', () => {
  let wrapper;
  const props = {
    index: 2,
    data: {
      id: 'abc1',
      slug: 'test-new',
      title: 'Some title',
      category: {
        name: 'Technology news',
        slug: 'technology-news',
      },
      thumbnail: 'image url',
    },
  };

  beforeAll(() => {
    wrapper = mount(<Item {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Item {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a thumbnail', () => {
    expect(wrapper.find('[data-test="item-thumbnail"]')).toHaveLength(1);
  });

  it('should have a category', () => {
    expect(wrapper.find('[data-test="item-category"]')).toHaveLength(1);
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="item-title"]')).toHaveLength(1);
  });

  it('should have an index', () => {
    expect(wrapper.find('[data-test="item-index"]')).toHaveLength(1);
  });
});
