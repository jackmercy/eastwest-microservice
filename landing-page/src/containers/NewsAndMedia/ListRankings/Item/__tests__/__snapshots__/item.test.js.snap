// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsAndMedia > ListRankings > Item should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test=""
>
  <div
    className="thumbnailLayout-0-2-2"
  >
    <div
      className="thumbnailWrapper-0-2-3"
    >
      <Link
        href="/news-and-media/technology-news/test-new"
        passHref={true}
      >
        <a
          href="/news-and-media/technology-news/test-new"
        >
          <img
            alt="Some title"
            className="thumbnail-0-2-4"
            data-test="item-thumbnail"
            loading="lazy"
            onError={[Function]}
            src="http://localhost:3411/files/public/image url"
          />
        </a>
      </Link>
    </div>
    <p
      className="ranking-0-2-5"
      data-test="item-index"
    >
      0
      <span>
        2
      </span>
    </p>
  </div>
  <div
    className="content-0-2-6"
  >
    <Link
      href="/news-and-media/technology-news"
      passHref={true}
    >
      <a
        className="category-0-2-7"
        data-test="item-category"
        href="/news-and-media/technology-news"
      >
        Technology news
      </a>
    </Link>
    <Link
      href="/news-and-media/technology-news/test-new"
      passHref={true}
    >
      <a
        className="title-0-2-8"
        data-test="item-title"
        href="/news-and-media/technology-news/test-new"
        title="Some title"
      >
        <Responsive(LinesEllipsis)
          basedOn="words"
          ellipsis="..."
          innerRef={[Function]}
          maxLine="3"
          text="Some title"
          trimRight={true}
        />
      </a>
    </Link>
  </div>
</div>
`;
